### PR TITLE
Add Snowplow analytics

### PR DESCRIPTION
**Pull request by arranfw**
- Makefile changes to make ENV_NAME available to FE
- Conditionally load a script based on the environment
- dev/test will load the dev snowplow script
- no script for prod yet

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -24,4 +24,5 @@ yarn-debug.log*
 yarn-error.log*
 
 # local env files
-.env*
\ No newline at end of file
+.env*
+public
```
Comment (will-freshworks) [inline .prettierignore:28]: Don't you need `public`?

File: pages/_app.tsx [hunk 2]
```diff
@@ -4,11 +4,29 @@ import Head from 'next/head';
...
+import Script from 'next/script';
 
 // TODO should be an ENV var
 const isMaintenanceMode = false;
 
+const envName = process.env.NEXT_PUBLIC_ENV_NAME;
+
+const getAnalyticsScriptPath = (envName?: string) => {
+  switch (envName) {
+    case 'dev':
```
Comment (will-freshworks) [inline pages/_app.tsx:16]: https://stackoverflow.com/questions/13207927/switch-statement-for-multiple-cases-in-javascript

File: pages/_app.tsx [hunk 3]
```diff
@@ -24,6 +42,8 @@ const Application = ({ Component, pageProps }: AppProps) => {
       </main>
 
       <Footer />
+
+      {analyticsScriptPath ? <Script src={analyticsScriptPath} /> : null}
```
Comment (will-freshworks) [inline pages/_app.tsx:45]: Never seen `next/script` before. Nicer than the react method I've used in the past: https://stackoverflow.com/questions/34424845/adding-script-tag-to-react-jsx

**Review by will-freshworks — APPROVED** [review]
Questions and pedantry. Also, the `public` thing confuses me because I thought you needed that for favicons and stuff.

**Comment by will-freshworks** [comment]
> Questions and pedantry. Also, the `public` thing confuses me because I thought you needed that for favicons and stuff.

Okay, so, I now see that this is `.prettierignore` and not `.gitignore`, so you can add me to `.arranignore`

File: .prettierignore (hunk 1 above)
Comment (arranfw) [reply .prettierignore:28]: Not for prettier!

File: pages/_app.tsx (hunk 2 above)
Comment (arranfw) [reply pages/_app.tsx:16]: Fair enough 😄

File: pages/_app.tsx (hunk 3 above)
Comment (arranfw) [reply pages/_app.tsx:45]: Vercel is so thoughtful <3

**Pull request merged**